projects/vdk-plugins/vdk-meta-jobs/src/vdk/plugin/meta_jobs/remote_data_job.py

```diff
@@ -0,0 +1,214 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import logging
+import os
+import time
+import uuid
+from enum import Enum
+from typing import Optional
+
+from taurus_datajob_api import ApiClient
+from taurus_datajob_api import Configuration
+from taurus_datajob_api import DataJobExecution
+from taurus_datajob_api import DataJobExecutionRequest
+from taurus_datajob_api import DataJobsExecutionApi
+from urllib3 import Retry
+from vdk.plugin.control_api_auth.authentication import Authentication
+
+log = logging.getLogger(__name__)
+
+
+class JobStatus(str, Enum):
+    """
+    Enum for the possible statuses a job execution can have.
+    """
+
+    SUBMITTED = "submitted"
+    RUNNING = "running"
+    SUCCEEDED = "succeeded"
+    CANCELLED = "cancelled"
+    SKIPPED = "skipped"
+    USER_ERROR = "user_error"
+    PLATFORM_ERROR = "platform_error"
+
+
+class RemoteDataJob:
```

Comment:
I see that this class has a lot of overlap with the class `VDKhook`. 
Do you plan on deduping the overlap?

Reply:
I don't like it as well. But I have not hit yet the https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)